### PR TITLE
[STORM-1293] port storm.messaging.netty-integration-test to java

### PR DESCRIPTION
**Pull request by nd368**

### CODE REVIEW

File: storm-core/test/jvm/org/apache/storm/messaging/NettyIntegrationTest.java [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+        daemonConf.put(Config.STORM_LOCAL_MODE_ZMQ, true);
+        daemonConf.put(Config.STORM_MESSAGING_TRANSPORT, "org.apache.storm.messaging.netty.Context");
+        daemonConf.put(Config.STORM_MESSAGING_NETTY_AUTHENTICATION, false);
+        daemonConf.put(Config.STORM_MESSAGING_NETTY_BUFFER_SIZE, 1024000);
+        daemonConf.put(Config.STORM_MESSAGING_NETTY_MIN_SLEEP_MS, 1000);
+        daemonConf.put(Config.STORM_MESSAGING_NETTY_MAX_SLEEP_MS, 5000);
+        daemonConf.put(Config.STORM_MESSAGING_NETTY_CLIENT_WORKER_THREADS, 1);
+        daemonConf.put(Config.STORM_MESSAGING_NETTY_SERVER_WORKER_THREADS, 1);
+
+        try (LocalCluster cluster = new LocalCluster.Builder().withSimulatedTime().withSupervisors(4)
```
Comment (Ethanlm) [inline storm-core/test/jvm/org/apache/storm/messaging/NettyIntegrationTest.java:58]: We can separate this to be 
```
Builder builder = new Builder()
.withSimulatedTime()
.withSupervisors(4)
.withSupervisorSlotPortMin(6710)
.withDaemonConf(daemonConf);

try (LocalCluster cluster = builder.build()) {
xxx
}

```

File: storm-core/test/jvm/org/apache/storm/messaging/NettyIntegrationTest.java [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+
+            MockedSources mockedSources = new MockedSources(Collections.singletonMap("1", testTuples));
+
+            CompleteTopologyParam completeTopologyParams = new CompleteTopologyParam();
+            completeTopologyParams.setStormConf(Collections.singletonMap(Config.TOPOLOGY_WORKERS, 3));
+            completeTopologyParams.setMockedSources(mockedSources);
+
+            Map<String, List<FixedTuple>> results = Testing.completeTopology(cluster, topology, completeTopologyParams);
+
+            assertThat(Testing.readTuples(results, "2").size(), is(6 * 4));
```
Comment (Ethanlm) [inline storm-core/test/jvm/org/apache/storm/messaging/NettyIntegrationTest.java:90]: Can we use `assertEquals` here?

**Comment by nd368** [comment]
changes made - thanks for review :)

**Comment by Ethanlm** [comment]
Travis failure is not related.

**Pull request merged**